### PR TITLE
Bug/sc 116617 event frame webid not consistent

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

**Review by liamlynch-data — COMMENTED** [review]
Did a first pass with a minor thing and a refactoring suggestion.
I can see the issues in the ticket are fixed and testing confirms that:
- if the recipe + batch mode are used it should not get the right event_frame_webid via the batching system (no possibility of a stale one) ✅ 
- event_frame_webid now present in the custom dataset schema returned when needed ✅ 

But when testing I seem to get the wrong number of distinct event_frame_webid on the dataset with no batching - to discuss

File: custom-recipes/pi-system-retrieve-event-frames/recipe.py [hunk 1]
```diff
@@ -123,7 +123,8 @@
                 item_row.update(item)
                 unnested_items_rows.append(item_row)
             if (not item) and ("Value" in base_row):
-                item_row = {"event_frame_webid": event_frame_webid}
+                # item_row = {"event_frame_webid": event_frame_webid}
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply custom-recipes/pi-system-retrieve-event-frames/recipe.py]: 👍 Done in [639a284](https://github.com/dataiku/dss-plugin-osisoft/pull/20/commits/639a284f2ff456bac8a5adf91a1ec1395674e572)

File: python-lib/osisoft_client.py [hunk 2]
```diff
@@ -78,32 +79,38 @@ def get_rows_from_webids(self, input_rows, data_type, start_date=None, end_date=
             url = self.endpoint.get_data_from_webid_url(endpoint_type, data_type, webid)
             requests_kwargs = self.generic_get_kwargs()
             requests_kwargs['url'] = url
-            batch.append(requests_kwargs)
+            web_ids.append(webid)
+            batch_requests_parameters.append(requests_kwargs)
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply python-lib/osisoft_client.py:83]: The batch endpoint can be used for any type of operations, not only recovering based on web ids, and in future we might want to use _batch_requests to speed up other transactions in the plugin...

[the remark this answers is not in the document]
Comment (liamlynch-data) [reply python-lib/osisoft_client.py:83]: ok - no problem to leave as is

File: python-connectors/pi-system_event-frames-search/connector.py [hunk 3]
```diff
@@ -149,7 +151,7 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
                         yield event_frame_copy
                         limit.add_record()
                         if limit.is_reached():
```
Comment (liamlynch-data) [inline python-connectors/pi-system_event-frames-search/connector.py:153]: I think we can/should probably remove this call to add_record() too?

Comment (alexbourret) [reply python-connectors/pi-system_event-frames-search/connector.py]: indeed ! [a0f8bb8](https://github.com/dataiku/dss-plugin-osisoft/pull/20/commits/a0f8bb8cda341f300c1be2729618a5913fa3625f)

Comment (liamlynch-data) [reply python-connectors/pi-system_event-frames-search/connector.py]: 👍

**Pull request merged**